src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -1209,7 +1209,9 @@ public override Task<ImplementationResult> ImplementationAsync(DelegatedPosition
         // servers.
         if (response?.Response is null or [{ Diagnostics: null }, ..])
         {
-            return null;
+            // Important that we send back an empty list here, because null would result it the above method throwing away any other
+            // diagnostics it receives from the other delegated server
+            return Array.Empty<VSInternalDiagnosticReport>();
```

Comment:
Why not just change line 1171 to allow null for html diagnostics?

Reply:
Because @ryanbrandenburg added it as part of getting diagnostic integration tests to pass, so there must be scenarios where it sometimes happens. Basically I'm assuming that he and I are both right, and the current break is just the confluence of our two code changes.

Reply:
🤷

works for me. Test would be nice but I don't think should block this (and I also don't remember how easy it is to mock out the HTML server for testing)

Reply:
There are tests... integration tests... that are skipped right now